From 537ba9d023fe494689b8c501abd37be16fcb35c8 Mon Sep 17 00:00:00 2001
From: Nikita Yushchenko <nikita.yoush@cogentembedded.com>
Date: Wed, 9 Mar 2022 15:12:15 +0300
Subject: [PATCH 029/151] arm64: dts: renesas: r8a779f0: remove references to
 gpio4

On S4 SoC gpio4 is only accessible from control domain.

Unless access is explicitly enabled by software running in control
domain, attempt to access the registers causes exception.

Signed-off-by: Nikita Yushchenko <nikita.yoush@cogentembedded.com>
---
 arch/arm64/boot/dts/renesas/r8a779f0-vc4.dts | 86 +-------------------
 1 file changed, 2 insertions(+), 84 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a779f0-vc4.dts b/arch/arm64/boot/dts/renesas/r8a779f0-vc4.dts
index e559df466b4d..e7b578eae1cc 100644
--- a/arch/arm64/boot/dts/renesas/r8a779f0-vc4.dts
+++ b/arch/arm64/boot/dts/renesas/r8a779f0-vc4.dts
@@ -68,11 +68,6 @@
 	clock-frequency = <32768>;
 };
 
-// Allowing &gpio4 to initialize causes boot hang
-&gpio4 {
-	status = "disable";
-};
-
 &i2c1 {
 	status = "okay";
 
@@ -199,9 +194,9 @@
 	clock-frequency = <100000000>;
 };
 
-/*
 &rswitch {
-	status = "disabled";
+	status = "okay";
+
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -241,83 +236,6 @@
 		};
 	};
 };
-*/
-
-&rswitch {
-	compatible = "renesas,rswitch2-r8a779f0",
-				     "renesas, rswitch2";
-	interrupts = <GIC_SPI 280 IRQ_TYPE_LEVEL_HIGH>,
-				 <GIC_SPI 281 IRQ_TYPE_LEVEL_HIGH>,
-				 <GIC_SPI 282 IRQ_TYPE_LEVEL_HIGH>,
-				 <GIC_SPI 283 IRQ_TYPE_LEVEL_HIGH>,
-				 <GIC_SPI 284 IRQ_TYPE_LEVEL_HIGH>,
-				 <GIC_SPI 285 IRQ_TYPE_LEVEL_HIGH>,
-				 <GIC_SPI 286 IRQ_TYPE_LEVEL_HIGH>,
-				 <GIC_SPI 287 IRQ_TYPE_LEVEL_HIGH>,
-				 <GIC_SPI 288 IRQ_TYPE_LEVEL_HIGH>,
-				 <GIC_SPI 289 IRQ_TYPE_LEVEL_HIGH>,
-				 <GIC_SPI 290 IRQ_TYPE_LEVEL_HIGH>,
-				 <GIC_SPI 291 IRQ_TYPE_LEVEL_HIGH>,
-				 <GIC_SPI 292 IRQ_TYPE_LEVEL_HIGH>,
-				 <GIC_SPI 293 IRQ_TYPE_LEVEL_HIGH>,
-				 <GIC_SPI 294 IRQ_TYPE_LEVEL_HIGH>,
-				 <GIC_SPI 295 IRQ_TYPE_LEVEL_HIGH>,
-				 <GIC_SPI 296 IRQ_TYPE_LEVEL_HIGH>,
-				 <GIC_SPI 297 IRQ_TYPE_LEVEL_HIGH>,
-				 <GIC_SPI 298 IRQ_TYPE_LEVEL_HIGH>,
-				 <GIC_SPI 299 IRQ_TYPE_LEVEL_HIGH>;
-	interrupt-names = "gwca0_rxtx0", "gwca0_rxtx1",
-					  "gwca0_rxtx2", "gwca0_rxtx3",
-					  "gwca0_rxtx4", "gwca0_rxtx5",
-					  "gwca0_rxtx6", "gwca0_rxtx7",
-					  "gwca1_rxtx0", "gwca1_rxtx1",
-					  "gwca1_rxtx2", "gwca1_rxtx3",
-					  "gwca1_rxtx4", "gwca1_rxtx5",
-					  "gwca1_rxtx6", "gwca1_rxtx7",
-					  "gwca0_rxts0", "gwca0_rxts1",
-					  "gwca1_rxts0", "gwca1_rxts1";
-		
-	status = "okay";
-
-
-	ports {
-		#address-cells = <1>;
-		#size-cells = <0>;
-		port@0 {
-			reg = <0>;
-			phy-mode = "sgmii";
-			phy-handle = <&etha0>;
-			#address-cells = <1>;
-			#size-cells = <0>;
-			etha0: ethernet-phy@0 {
-				reg = <3>;
-				compatible = "ethernet-phy-id002B.0218";
-			};
-		};
-		port@1 {
-			reg = <1>;
-			phy-mode = "sgmii";
-			phy-handle = <&etha1>;
-			#address-cells = <1>;
-			#size-cells = <0>;
-			etha1: ethernet-phy@1 {
-				reg = <3>;
-				compatible = "ethernet-phy-id002B.0218";
-			};
-		};
-		port@2 {
-			reg = <2>;
-			phy-mode = "sgmii";
-			phy-handle = <&etha2>;
-			#address-cells = <1>;
-			#size-cells = <0>;
-			etha2: ethernet-phy@2 {
-				reg = <3>;
-				compatible = "ethernet-phy-id002B.0218";
-			};
-		};
-	};
-};
 
 &scif0 {
 	status = "okay";
-- 
2.17.1

